pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -657,3 +655,30 @@ func getNodeMAC(node *corev1.Node) (net.HardwareAddr, error) {
 	}
 	return mac, nil
 }
+
+func (c *Controller) getNodeIPForPods(node *corev1.Node) (net.IP, error) {
```

Comment:
I think the method can just be `getNodeTransportAddress` as it's just NodeIP will be used as transport IP address if it's not specified explicitly. `getNodeIPForPods` sounds different meaning.